crates/kitsune_p2p/fetch/src/backoff.rs

```diff
@@ -0,0 +1,110 @@
+use tokio::time::{Duration, Instant};
+
+use backon::{BackoffBuilder, FibonacciBackoff, FibonacciBuilder};
+
+/// The number of times to retry a fetch before giving up.
+pub const BACKOFF_RETRY_COUNT: usize = 10;
+
+/// A backoff strategy for use when fetching data from remote nodes that appear to not be responding.
+#[derive(Debug)]
+pub struct FetchBackoff {
+    backoff: FibonacciBackoff,
+    current_wait: Duration,
+    started_wait_at: Instant,
+    expired: bool,
+}
+
+impl FetchBackoff {
+    /// Create a new instance with the given initial delay.
+    pub fn new(initial_delay: Duration) -> Self {
+        let mut backoff = FibonacciBuilder::default()
+            .with_jitter()
+            .with_min_delay(initial_delay)
+            .with_max_delay(6 * initial_delay)
```

Comment:
This is relatively high and with retries it will result in quite a long total wait time to notice a source is gone. I've been deliberately conservative here to avoid stepping too far from the current implementation in one go. It means if somebody is offline for 5-10 minutes, we will still try to fetch from them when they come back. I'd like to make this a bit more aggressive if it works.

Reply:
I did the math, I think the total wait would be `30 + 30 + 60 + 90 + 150 + 240 + 300 + 300 + 300 + 300 = 1800` which is 30 minutes. Also I checked the source of FibonacciBackoff, and it bases the jitter on the min_delay, so each retry actually adds anywhere from 0-30 seconds to the next retry interval, so with current settings on average we can expect an extra 15 seconds * 10 = 2.5 minutes from jitter. 

So 32.5 minutes. That seems too lax, given that even the agent info itself expires after 20 min, after which it would definitely be impossible to reach the source. Maybe this should be tuned to be something closer to 20 min total, as an upper limit?

With the current settings, a min_delay of 10 sec puts the total sum of delays at 19 min = `10 + 10 + 20 + 30 + 50 + 80 + 130 + 210 + 300 + 300`, but maybe that's too quick for the first few retries? 

Maybe 30 sec min_delay with 8 retries? That's exactly 20 min.

Reply:
That sounds good! I wouldn't want to bring it down super low, like below 5 minutes, because I think that's a reasonable amount of time for a source to be busy but 20 minutes sounds better than 30+ to me. I've changed these to the values you've suggested.

Thanks for doing the maths on this, I did some rough calculations a few times as I went here but now it's been done properly it's probably worth a code comment. I'll find somewhere to add one